controllers/AdminController.php

```diff
@@ -301,16 +308,26 @@ public function actionInfo($id)
      */
     public function actionSwitch($id)
     {
-        $user = $this->findModel($id);
+        $old = Yii::$app->user;
+        $session = Yii::$app->session;
 
-        $old = \Yii::$app->user;
+        if($id == 'previous' && $session->has('previous_user')) {
```

Comment:
I'd recommend using a constant for `'previous'` and `'previous_user'`. Wouldn't call it previous, rather "original_user_id" or something.